service/prometheus/scrapeconfig.go

```diff
@@ -24,8 +24,15 @@ const (
 )
 
 // getJobName takes a cluster ID, and returns a suitable job name.
-func getJobName(service v1.Service) string {
-	return fmt.Sprintf("%s-%s", jobNamePrefix, service.Namespace)
+func getJobName(service v1.Service, name string) string {
+	jobName := fmt.Sprintf("%s-%s", jobNamePrefix, service.Namespace)
+
+	// TODO: Remove once all job names have suffixes.
```

Comment:
Instead of having one job for guest cluster scraping, I wanna break it down into specific area (e.g: apiserver, kubelets). This simplifies the configuration a lot.